pkg/web/static/js/counterparties.js

```diff
@@ -21,12 +21,36 @@ document.body.addEventListener('htmx:afterRequest', (e) => {
   }
 });
 
+// Use SlimSelect to create a searchable select dropdown for countries in the add counterparty modal form.
 const countrySelect = new SlimSelect({
   select: '#countries',
   settings: {
     contentLocation: document.getElementById('country-content'),
   },
 });
-
 countriesArray.unshift({ 'placeholder': true, 'text': 'Select a country', 'value': '' });
-countrySelect.setData(countriesArray);
\ No newline at end of file
+countrySelect.setData(countriesArray);
+
+// Set the country value in the edit counterparty modal form.
+document.body.addEventListener('htmx:afterSettle', (e) => {
+  const cpartyID = document.getElementById('cparty_id');
+  const cpartyPreviewEP = `/v1/counterparties/${cpartyID?.value}/edit`;
+
+  if (e.detail.requestConfig.path === cpartyPreviewEP && e.detail.requestConfig.verb === 'get') {
+    // Initialize SlimSelect for the country dropdown in the edit counterparty modal form.
+    const countrySelect = new SlimSelect({
+      select: '#country',
+      settings: {
+        contentLocation: document.getElementById('cparty_modal'),
```

Comment:
It seems that adding an empty div for the content was too specific and setting the location to the modal solves the issue I was running into where the select options weren't appearing inside modals.

Reply:
nice!